### PR TITLE
Fixed - System email template access issue

### PR DESCRIPTION
**Pull request by Dillon-Brown**
## Description

This change adds a new type "System" to email templates and assigns this type to system generated emails. This type cannot be edited by anyone other than an admin user.

## How To Test This

1. Check that system created email templates are set to type "system" on install.
2. Try to edit an email template with type "System" as non-admin.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [X] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/EmailTemplates/EditView.php [hunk 1]
```diff
@@ -1,6 +1,48 @@
...
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
 
+global $current_user;
```
Comment (daniel-samson) [inline modules/EmailTemplates/EditView.php]: Multiple declarations of global $current_user in the file. Consider removing the duplicates.

Comment (Dillon-Brown) [inline modules/EmailTemplates/EditView.php]: Fixed

File: install/language/en_us.lang.php [hunk 2]
```diff
@@ -468,6 +468,7 @@
 
 	'advanced_password_new_account_email' => array(
 		'subject' => 'New account information',
+		'type' => 'system',
```
Comment (horus68) [inline install/language/en_us.lang.php]: Why not initial cap for "System"?

**Comment by Mausino** [comment]
Is it patch for issue #3852 ??? or is only restriction, which allow to edit the system templates only by admin ?

**Comment by Dillon-Brown** [comment]
@Mausino This PR is unrelated to #3852. It just adds a new email template type that restricts edit access to admin only.

**Pull request merged**